apis/pkg/v1alpha1/interfaces.go

```diff
@@ -140,6 +143,19 @@ func (p *Provider) SetRevisionHistoryLimit(l int64) {
 	p.Spec.RevisionHistoryLimit = &l
 }
 
+// GetIgnoreCrossplaneConstraints of this Provider.
+func (p *Provider) GetIgnoreCrossplaneConstraints() bool {
+	if p.Spec.IgnoreCrossplaneConstraints == nil {
+		return false
+	}
```

Comment:
Nit: I'm a little wary of these methods automatically handling the conversion to and from a bool pointer. It seems convenient and I'm sure is pretty safe in this context, but it could be a little surprising to users of the (Go) API given that AFAIK we don't do this in our interface getters and setters anywhere else.

Reply:
That seems like a reasonable concern. I really was struggling to see why we really even need this field to be a `*bool` instead of a `bool`, but was trying to follow API conventions. Note that we also can do defaulting with v1 CRDs so there may be some areas where we have traditionally used a pointer because we wanted to default to a value that is not the zero value of the type (so we needed to distinguish between zero value and no value) that could be updated.